### PR TITLE
Muon - Embed Widgets into Interface

### PR DESCRIPTION
**Pull request by TWJubb**
Refs #23642 

This is the final PR of the set, embedding the widgets for the home tab, grouping widget and load widgets into the muon analysis GUI. 

There is an ammendment to the tabbing which will be justified shortly.

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AnthonyLim23** [comment]
you have some failing unit tests

File: scripts/Muon/GUI/Common/ADSHandler/workspace_naming.py [hunk 1]
```diff
@@ -7,37 +7,37 @@
 from __future__ import (absolute_import, division, print_function)
 
 
-def get_raw_data_workspace_name(context):
-    return context._base_run_name() + "_raw_data"
+def get_raw_data_workspace_name(context, run):
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/ADSHandler/workspace_naming.py]: is run the run number or the ws for that run?

File: scripts/Muon/GUI/Common/dock/dockable_tabs.py [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+        Remove the tab with the given name, even if it is detached.
+        :param name: the name of the tab to be removed.
+        """
+        # Remove the tab if it is attached
+        attached = False
+        for index in xrange(self.count()):
+            if str(name) == str(self.tabText(index)):
+                self.removeTab(index)
+                attached = True
+                break
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/dock/dockable_tabs.py]: can we do this without the break?

File: scripts/Muon/GUI/Common/dock/dockable_tabs.py [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+
+        # If the tab is not attached, close it's window and
+        # remove the reference to it
+        if not attached:
+            for key in self.detachedTabs:
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/dock/dockable_tabs.py]: won't this have problems on python3 without the list(...)?

File: scripts/Muon/GUI/Common/dock/dockable_tabs.py [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+                    return False
+
+        def closeEvent(self, _event):
+            """
+            If the window is closed, emit the onCloseSignal and give the
+            content widget back to the DetachableTabWidget.
+            """
+            self.onCloseSignal.emit(self.contentWidget, self.objectName(), self.windowIcon())
+
+    class TabBar(QtGui.QTabBar):
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/dock/dockable_tabs.py]: can we split these classes into their own files?

**Comment by AnthonyLim23** [comment]
I have found :bug: (I think you are fixing some else where)

1. It automatically gives me a non muon instrument
2. If I load the current run and then turn on the dead time table, the data does not seem to update
3. After doing number 2, clicking back throws loads of errors to the logger
4. I cannot get the group asymmetries 
5. Changing the time zero and first good data should automatically update all of the data
6. Rebin should automatically update the data (doesnt seem to do anything)
7. in the run info it is `MEv` not `MeV`
8. I opened muon analysis (1) and loaded some MUSR data. I then went into the new GUI and set dead time table and selected from table workspace. I picked the TFNorm table and the error message was not helpful as the end seemed to be cut off.
9. For the dead time table, I picked from other file and clicked Browse (a line edit might be better) and I do not know if it loaded it or not (there is no user feedback). I think it threw an error to the logger.
10. Does the group/pair option do anything?
11.  If I use browse it does not update the instrument name in the GUI or the browser
12. Loading HIFI 84447 (multi period data) throws lots of errors to the logger
13. I cannot seem to get it to period arithmetic
14. Changing the detector grouping and clicking update does not seem to do anything if I have loaded simultaneous or single data

File: scripts/Muon/GUI/Common/home_runinfo_widget/home_runinfo_widget_model.py [hunk 5]
```diff
@@ -44,16 +44,16 @@ def get_counts_in_MeV(self, counts):
...
         else:
             return 'Good frames not defined'
 
     def get_counts_per_good_frame_per_detector(self, counts):
         good_frames = self.get_log_value("goodfrm")
 
-        if good_frames is not 'Log not found':
-            return counts/good_frames/self._data.num_detectors()
+        if good_frames != 'Log not found':
+            return counts/float(good_frames)/self._data.num_detectors
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/home_runinfo_widget/home_runinfo_widget_model.py]: I think it should be `float(good_frames*self._data.num_detectors)` as I expect the number of detectors to be an int

File: scripts/Muon_Analysis_2.py [hunk 6]
```diff
@@ -64,22 +67,43 @@ def __init__(self, parent=None):
...
+        vertical_layout.addWidget(self.load_widget.load_widget_view)
+        vertical_layout.addWidget(self.tabs)
+        vertical_layout.addWidget(self.help_widget.view)
+        central_widget.setLayout(vertical_layout)
 
-        self.setCentralWidget(splitter)
+        self.setCentralWidget(central_widget)
         self.setWindowTitle("Muon Analysis version 2")
 
+        self.home_tab.group_widget.pairAlphaNotifier.add_subscriber(self.grouping_tab_widget.group_tab_presenter.loadObserver)
```
Comment (AnthonyLim23) [inline scripts/Muon_Analysis_2.py]: can we have a blank line between these please, just to make it easier to read

File: scripts/test/Muon/load_run_widget/loadrun_presenter_current_run_test.py [hunk 7]
```diff
@@ -128,24 +128,11 @@ def test_load_current_run_clears_previous_data_if_load_succeeds(self):
...
         self.wait_for_thread(self.presenter._load_thread)
 
-        self.assertEqual(self.view.get_run_edit_text(), "9999 (CURRENT RUN)")
+        self.assertEqual(self.view.get_run_edit_text(), "[9999] (CURRENT RUN)")
         self.assertEqual(self.presenter.filenames, ["9999.nxs"])
-        self.assertEqual(self.presenter.runs, [9999])
+        self.assertEqual(self.presenter.runs, [[9999]])
         self.assertEqual(self.presenter.workspaces, [[2]])
 
-    @run_test_with_and_without_threading
```
Comment (AnthonyLim23) [inline scripts/test/Muon/load_run_widget/loadrun_presenter_current_run_test.py]: I think this is a good test to have. Is there a reason to remove it?

File: scripts/test/Muon/load_file_widget/loadfile_model_test.py [hunk 8]
```diff
@@ -8,19 +8,23 @@
...
+        self.context = MuonDataContext()
+        self.context.instrument = 'EMU'
+        self.model = BrowseFileWidgetModel(self.data, self.context)
 
-        patcher = mock.patch('Muon.GUI.Common.load_file_widget.model.load_utils')
-        self.addCleanup(patcher.stop)
-        self.load_utils_patcher = patcher.start()
+        # patcher = mock.patch('Muon.GUI.Common.load_file_widget.model.load_utils')
+        # self.addCleanup(patcher.stop)
+        # self.load_utils_patcher = patcher.start()
```
Comment (AnthonyLim23) [inline scripts/test/Muon/load_file_widget/loadfile_model_test.py]: should we just delete these?

Comment (Mantid-Matthew) [reply scripts/test/Muon/load_file_widget/loadfile_model_test.py]: Yes we should

**Comment by Mantid-Matthew** [comment]
I have now addressed 1,8, 12, 15 and 16 mentioned above. 

4. is being dealt with in #24646
6. and 6. are being dealt with in #24675

I have also opened an issue to refactor the tabbing as discussed #24678

**Comment by AnthonyLim23** [comment]
I still get bug 15.

**Comment by AnthonyLim23** [comment]
got a new bug. If you open the GUI and then close it. You can then no longer reopen it as it is considered to be still open by Mantid and so it tries to rise it.

**Review by AnthonyLim23 — APPROVED** [review]
unrelated build failure

**Comment by AnthonyLim23** [comment]
@mantidproject/gatekeepers could we get this merged please?

**Pull request merged**